engine/src/main/scala/cromwell/server/CromwellRootActor.scala

```diff
@@ -108,16 +117,40 @@ import scala.language.postfixOps
 
   lazy val jobExecutionTokenDispenserActor = context.actorOf(JobExecutionTokenDispenserActor.props)
 
-  def abortJobsOnTerminate: Boolean
-
   lazy val workflowManagerActor = context.actorOf(
     WorkflowManagerActor.props(
       workflowStoreActor, ioActor, serviceRegistryActor, workflowLogCopyRouter, jobStoreActor, subWorkflowStoreActor, callCacheReadActor, callCacheWriteActor,
-    dockerHashActor, jobExecutionTokenDispenserActor, backendSingletonCollection, abortJobsOnTerminate, serverMode),
+      dockerHashActor, jobExecutionTokenDispenserActor, backendSingletonCollection, serverMode),
     "WorkflowManagerActor")
 
+  if (gracefulShutdown) {
+    // If abortJobsOnTerminate is true, aborting all workflows will be handled by the graceful shutdown process
+    CromwellShutdown.registerShutdownTasks(
+      abortJobsOnTerminate,
+      actorSystem = context.system,
+      workflowManagerActor = workflowManagerActor,
+      logCopyRouter = workflowLogCopyRouter,
+      jobStoreActor = jobStoreActor,
+      workflowStoreActor = workflowStoreActor,
+      subWorkflowStoreActor = subWorkflowStoreActor,
+      callCacheWriteActor = callCacheWriteActor,
+      ioActor = ioActor,
+      dockerHashActor = dockerHashActor,
+      serviceRegistryActor = serviceRegistryActor,
+      materializer = materializer
+    )
+  } else if (abortJobsOnTerminate) {
+    // If gracefulShutdown is false but abortJobsOnTerminate is true, set up a classic JVM shutdown hook
+    sys.addShutdownHook {
+      Try(Await.result(gracefulStop(workflowManagerActor, AbortTimeout, AbortAllWorkflowsCommand), AbortTimeout)) match {
+        case Success(_) => logger.info("All workflows aborted")
+        case Failure(f) => logger.error("Failed to abort workflows", f)
+      }
+    }
+  }
+
   override def receive = {
-    case _ => logger.error("CromwellRootActor is receiving a message. It prefers to be left alone!")
+    case _ => logger.error("Unknown message received by CromwellRootActor")
```

Comment:
Although in this case since the `receive` method has to be overridden and we really don't expect the CromwellRootActor to receive *any* message I think it makes sense to keep this as a logged error